src/main/java/Jason.java

```diff
@@ -0,0 +1,208 @@
+import jason.tasks.Task;
+import jason.tasks.Todo;
+import jason.tasks.Deadline;
+import jason.tasks.Events;
+
+import jason.errorhandling.JasonException;
+
+
+import java.util.Scanner;
+
+/**
+ * Represents the main class for the application Duke.
+ */
+public class Jason {
+    private static final int MAX_SIZE = 100;
+    private static Task[] list = new Task[MAX_SIZE];
+    private static int num = 0;
+
+
+    private static void markTask(String[] input) throws JasonException {
+        try {
+            int taskNumber = Integer.parseInt(input[1]) - 1;
+            if (taskNumber < 0 || taskNumber >= num) {
+                throw new JasonException("That task number is not valid. The list only has " + num + " task(s) currently");
+            }
+            list[taskNumber].markAsDone();
+            System.out.println("Nice! I've marked this task as done:");
+            System.out.println(list[taskNumber]);
+        } catch (NumberFormatException e) {
+            throw new JasonException("Please enter a valid task number.");
+        }
+    }
+
+    private static void unmarkTask(String[] input) throws JasonException {
+        try {
+            int taskNumber = Integer.parseInt(input[1]) - 1;
+            if (taskNumber < 0 || taskNumber >= num) {
+                throw new JasonException("That task number is not valid. The list only has " + num + " task(s) currently");
+            }
+
+            list[taskNumber].markAsNotDone();
+            System.out.println("OK, I've marked this task as not done yet:");
+            System.out.println(list[taskNumber]);
+        } catch (NumberFormatException e) {
+            throw new JasonException("Please enter a valid task number.");
+        }
+    }
+
+
+    private static void showTaskNumber() {
+        if (num == 1) {
+            System.out.println("Now you have 1 task in the list");
+        } else {
+            System.out.println("Now you have " + num + " tasks in the list");
+        }
+
+    }
+
+    private static void showList() throws JasonException {
+        if (num == 0) {
+            throw new JasonException("The task list is currently empty. Add some tasks!");
+        }
+
+        System.out.println("Here are the tasks in your list:");
+        for (int i = 0; i < num; i++) {
+            System.out.println(i + 1 + ":" + list[i]);
+        }
+    }
+
+    private static void todoTasks(String task) throws JasonException {
+        String taskDescription = task.substring(5).trim(); // trim() removes leading and trailing spaces
+        if (taskDescription.isEmpty()) {
+            throw new JasonException("The description of a todo cannot be empty.");
+        }
+        list[num] = new Todo(taskDescription);
+        System.out.println("Got it. I've added this task:");
+        System.out.println(list[num].toString());
+        num++;
+        showTaskNumber();
+    }
+
+    private static void deadlineTasks(String task) throws JasonException {
```

Comment:
for this and the next function: try keeping the level of abstraction same throughout the code, so maybe you can extract parts of the code into different functions and maintain a level of abstraction. 
https://nus-cs2113-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html#slap-hard